pkg/controller/secret/secret_controller.go

```diff
@@ -69,25 +66,66 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 	if err != nil {
 		return errors.Wrapf(err, "could not create the controller: %v", ControllerName)
 	}
-	predicateFilter := predicate.Funcs{
-		// get update event only when secret data is changed
-		UpdateFunc: func(e event.UpdateEvent) bool {
-			if string(e.ObjectOld.(*core.Secret).Data["userData"][:]) != string(e.ObjectNew.(*core.Secret).Data["userData"][:]) {
-				return true
-			}
-			return false
-		},
+
+	// We are watching multiple namespaces, so this will return the namespace defined within the WMCO CSV, the one that
+	// the operator is deployed within.
+	namespace, err := k8sutil.GetWatchNamespace()
+	if err != nil {
+		return errors.Wrap(err, "failed to get watch namespace")
 	}
 
-	// Watch for changes to primary resource UserDataSecret
-	err = c.Watch(&source.Kind{Type: &core.Secret{ObjectMeta: meta.ObjectMeta{Namespace: userDataNamespace,
-		Name: userDataSecret}}}, &handler.EnqueueRequestForObject{}, predicateFilter)
+	// Check that the private key exists, if it doesn't, log a warning
+	_, err = secrets.GetPrivateKey(kubeTypes.NamespacedName{Namespace: namespace, Name: secrets.PrivateKeySecret}, mgr.GetClient())
+	if err != nil {
+		log.Error(err, "Unable to retrieve private key, please ensure it is created")
+	}
+
+	// Watch for changes to the userData secret and enqueue the cloud-private-key if changed
+	err = c.Watch(&source.Kind{Type: &core.Secret{}},
+		&handler.EnqueueRequestsFromMapFunc{ToRequests: newUserDataMapper(namespace)},
```

Comment:
nit: You don't need this enqueuing as technically you would like to react upon the private secret change anyways.